src/ethereum/mod.rs

```diff
@@ -37,12 +43,41 @@ pub struct Ethereum {
     polling_interval: Duration,
     /// A handle to the event loop that runs mosaic.
     event_loop: Box<tokio_core::reactor::Handle>,
+    /// List of block observers. These are notified when any new block is generated.
+    observers: Vec<Reactor>,
+}
+
+/// This enum represents all reactors which will react to block generation.
+#[derive(Debug, Clone)]
+pub enum Reactor {
+    BlockReporter { block_store_address: Address, validator_address: Address },
```

Comment:
Looks like a candidate for the observer module instead of the ethereum module 🤔

Reply:
Let's talk about this in a call.
We should probably change `observe` trait to `react` 🤔

Reply:
Create `reactor.rs` module in `src`